app/models/concerns/profile_policy_scoring.rb

```diff
@@ -3,10 +3,12 @@
 # Methods that are related to computing the score of a profile or a policy
 module ProfilePolicyScoring
   def score(host: nil)
-    results = test_results.latest
-    results = results.where(host: host) if host
-    return 0 if results.blank?
+    return super().to_f unless host
```

Comment:
How does this work? The `super` is supposed to be the `score` field on `Profile` which this module method overrides?

```
> Profile.first.score
  Profile Load (1.6ms)  SELECT  "profiles".* FROM "profiles" ORDER BY "profiles"."id" ASC LIMIT $1  [["LIMIT", 1]]
NoMethodError: super: no superclass method `score' for #<Profile:0x00007f45c88582b0>
from /home/vkrizan/.gem/gems/activemodel-5.2.4.5/lib/active_model/attribute_methods.rb:427:in `method_missing'
```

Reply:
I'm trying to keep compatibility with the old `score(host)` method. If the `host` argument is not passed, we're falling back to the DB column, otherwise we're calculating the score for only the given host.

Reply:
Nit: parenthesis are unnecessary in ruby :)

Reply:
Nope, `super()` and `super` aren't the same things in Ruby

Reply:
Could you explain? Does `super` forward params and `super()` not forward params or something?

Reply:
Yes, as you said

Reply:
This is super :smile:

Reply:
I was interested in the method resolution that is done here. `ProfilePolicyScoring` is a module that is being included in a class, that overrides the dynamically added method (added by Rails). This looks like a composition rather than an inheritance, hence my curiosity.

Reply:
Ruby is magic :laughing: